core/src/syn/v1/stmt/relate.rs

```diff
@@ -62,7 +62,7 @@ fn relate_oi(i: &str) -> IResult<&str, (Value, Value, Value)> {
 
 fn relate_o(i: &str) -> IResult<&str, (Value, Value)> {
 	let (i, _) = mightbespace(i)?;
-	let (i, kind) = alt((into(thing), into(table)))(i)?;
+	let (i, kind) = alt((into(thing), into(table), into(param), into(subquery)))(i)?;
```

Comment:
There is an inconsistency here between the old parser and the new. The old parser will allow you to do bare statements like `SELECT foo FROM bar->rel->RETURN baz;` while the new requires that you surround it with `()`. I think the parser here should also require `()`. 

Also do we want to allow general statements in this position? Is `RELATE (DEFINE TABLE foo)->bar->(REMOVE TABLE faz);` syntax we want to support?

Reply:
doesn't `DEFINE TABLE foo` return `NONE` though so that query wouldn't work, though I think it is syntactically valid if semantically incorrect. and I think the same thing should be true of the kind

Reply:
Cool, that's how it works at the moment

Reply:
I just tested but actually this works, this is already with parser2:
```sql
relate (create test)->edge->create test
```

Reply:
This actually already seems to work in current versions of SurrealDB...
<img width="1081" alt="image" src="https://github.com/surrealdb/surrealdb/assets/17960250/e2df329e-81ac-4aec-b577-219a22592b76">